File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java

// [7,1,5,3,6,4]

// min : 1
// diff: 5
class Solution {
    public int maxProfit(int[] prices) {
        int minV = Integer.MAX_VALUE;
        int diff = 0;
        for(int i = 0;i<prices.length;i++){
            minV = Math.min(minV, prices[i]);
            diff = Math.max(diff, prices[i] - minV);
        }
        return diff;
        
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    currMin := int(^uint(0) >> 1)
    result := 0
    for _,value := range prices{
        if value < currMin{
            currMin = value
        }
        if value - currMin > result{
            result = value - currMin
        } 
    }
    return result
}